packages/puppeteer/src/utils.ts

```diff
@@ -76,7 +76,16 @@ export async function assertFrameReady(frame: Frame): Promise<void> {
   // Check if the page is loaded.
   let pageReady = false;
   try {
-    pageReady = await frame.evaluate(pageIsLoaded);
+    // Puppeteer freezes on unloaded iframes. Set a race timeout in order to handle that.
+    // @see https://github.com/dequelabs/axe-core-npm/issues/727
+    const timeoutPromise = new Promise(resolve => {
+      setTimeout(() => {
+        resolve('timeout');
```

Comment:
Resolving with the value "timeout" is not at all idiomatic. If we time out, that is a sign of a problem, which is what the `reject` callback is meant to be used for. If we reject this promise (as discussed), we'd enter the `catch` block below (which _should_ throw the `Page/Frame is not ready` error).

Reply:
we do not want to throw in these scenarios

Reply:
We do throw in these scenarios. See below.